### DISCUSSION TITLE
Library constantly updating

### DISCUSSION THREAD

**Discussion by Bminor7b5** [rern/rAudio#713] [Bug Reports]
Using a RaspPi 3 B with all music files on USB harddrive.

Since downloading the latest 64bit image the Library will randomly update sometimes in the middle of playing an album/playlist. The Library goes from having 2,000+ artists to having none. 
The USB disk drive is sometimes connected/disconnected, given a different name and then goes into update mode without actually updating. (I've left it overnight and it will still be going)

I tried updating, I've gone back to older images which doesn't seem to help, I've checked the drive (which is relatively new and only ever used for storing music files and was working fine with rAudio previously) I've tried unmounting, remounting, forget drive - nothing seems to make any difference. 
So at the moment I can't access any of my music from the harddrive (or I can play a handful of songs after rebooting and restoring the library before it will go into a never-ending library update)

Any suggestions would be much appreciated

![Screenshot_20211222-223630_RuneAudio](https://user-images.githubusercontent.com/96576121/147235104-32efa044-1a70-4785-b96e-c77a1a4d5fe1.jpg)
![Screenshot_20211223-094840_RuneAudio](https://user-images.githubusercontent.com/96576121/147235107-becae78f-b82a-4c32-9e99-94ab159cb1da.jpg)

**Comment by rern** [comment]
Plug and Play USB drives will be scanned to update Library database automatically.
Intermittent connect/disconnect indicates hardware problems.
Try:
- Switch to other 3 USB ports
- Change USB cable
- On another RPi if available

Also observe the red LED on RPi for any blinks which indicates under voltage power supply.

**Reply by Bminor7b5** [reply]
Thanks Rern,
I have since updated to the 20211222 version and initially seems to be working well. Won't get a chance to test properly until after Christmas. Had tried the other ports and using the official pi psu but will try those other suggestions if there's still issues. 
Thanks for all your work, sound quality is so much better than kodi

**Reply by Bminor7b5** [reply]
For anyone else having similar issues who finds this thread: the hard drive was faulty. It was the first thing I suspected but had initially ruled it out as I had checked it on my windows pc and run chkdsk and all seemed fine. The drive was less than 3 years old and only used for playing music files on my pi so I was inclined to think the problem was elsewhere especially as it had been working fine until I updated Rune. After trying all the other suggestions from Rern and a short period of stability the drive started acting up again so I did a more rigorous test and sure enough it was definitely faulty...so new harddrive and all is well
